src/software/ai/hl/stp/tactic/cherry_pick_tactic.cpp

```diff
@@ -52,7 +53,12 @@ void CherryPickTactic::calculateNextAction(ActionCoroutine::push_type& yield)
     } while (true);
 }
 
-void CherryPickTactic::accept(TacticVisitor& visitor) const
+void CherryPickTactic::accept(NonMutableTacticVisitor& visitor) const
```

Comment:
Pretty sure the argument needs to be a `const NonMutableTacticVisitor& visitor`. This goes for all tactics.

Reply:
This seems to create issues in MotionConstraintManager due to it not being to update the set of whitelisted constraints

Reply:
The `MotionContstraintsManager` should inherit the `MutableTacticVisitor` then since it's modifying the tactics to change the whitelists.